### DISCUSSION TITLE
A strange poor performance on a small amount of vectors?

### DISCUSSION THREAD

**Discussion by yuGAN6** [milvus-io/milvus#14049] [Q&A and General discussion] [answered]
### Just installed  v2.0.0rc8 following the tutorials, tried searching on a little BERT-encoded vectors, and got an about 300ms performance on searching. Here's some details:

OS: CentOS Linux release 7.6.1810
GPU: nvidia tesla P-100
memory: 64g
milvus verison (docker): milvusdb/milvus:v2.0.0-rc8-20211104-d1f4106     ( standalone )
SDK: pymilvus-v2.0.0rc8
All the other configs remains default.

 Num. of entities: 10500
 Vector dimensions: 768,  single precision float.

Params tried:
Index type : FLAT, IVF_FLAT, IVF_SQ8
nlist : 1024, 2048
nprobe: 32, 64
metric_type: IP
topK(limit): 1

Interesting that whichever type of index I create, the searching performance remains the same around 300 ms. 
I tried reloading the collection every time after create_index() but it remains like that.
Or Is it possible that the index were not created?

**Comment by yhmo** [comment, marked as answer]
The milvus 2.0 use timestamp to sync actions across nodes and keep data consistent:
https://github.com/milvus-io/milvus/blob/2a6ada75367e54a8b498e69aba3c70700ed55705/docs/developer_guides/how-guarantee-ts-works.md

By default, for each 200ms the system will broadcast a system time to all nodes to make sure actions within the short time can be consumed successfully.
When a search request comes in, it wait until a system time is triggered so that this request can "see" the latest data. So you observed the search performance is about 200 ~ 400ms.
The search() API has a parameter named "guarantee_timestamp":
```
    def search(self, collection_name, data, anns_field, param, limit, expression=None, partition_names=None,
               output_fields=None, timeout=None, round_decimal=-1, **kwargs):
        """
        Searches a collection based on the given expression and returns query results.
        ......
        :param kwargs:
            * *guarantee_timestamp* (``int``) --
              This function instructs Milvus to see all operations performed before a provided timestamp. If no
              such timestamp is provided, then Milvus will search all operations performed to date.
        """
```
Set the "guarantee_timestamp" = 1 can skip the waiting. Search request can be executed immediately.

**Reply by yuGAN6** [reply]
It works, thanks!  That helps a lot for the consistency on the distributed system. Great job!
